tests/ha/filesystem.pm

```diff
@@ -86,27 +86,40 @@ sub run {
 
     if (is_node(1)) {
         # Create the Filesystem resource
-        assert_script_run
-"EDITOR=\"sed -ie '\$ a primitive $fs_rsc ocf:heartbeat:Filesystem params device='$fs_lun' directory='/srv/$fs_rsc' fstype='$fs_type''\" crm configure edit", $default_timeout;
-
+        my $clean_flag = undef;
+        my $edit_crm_config_script = "#!/bin/sh
+EDITOR='sed -ie \"\$ a primitive $fs_rsc ocf:heartbeat:Filesystem params device=\'$fs_lun\' directory=\'/srv/$fs_rsc\' fstype=\'$fs_type\'\"' crm configure edit
+";
         # Only OCFS2 can be cloned
         if ($fs_type eq 'ocfs2') {
-            assert_script_run "EDITOR=\"sed -ie 's/^\\(group base-group.*\\)/\\1 $fs_rsc/'\" crm configure edit", $default_timeout;
+            $edit_crm_config_script .= "
+EDITOR='sed -ie \"s/^\\(group base-group.*\\)/\\1 $fs_rsc/\"' crm configure edit
+";
         }
         else {
             if ($resource eq 'drbd_passive') {
-                assert_script_run "EDITOR=\"sed -ie '\$ a colocation colocation_$fs_rsc inf: $fs_rsc ms_$resource:Master'\" crm configure edit", $default_timeout;
-                assert_script_run "EDITOR=\"sed -ie '\$ a order order_$fs_rsc Mandatory: ms_$resource:promote $fs_rsc:start'\" crm configure edit", $default_timeout;
+                $edit_crm_config_script .= "
+EDITOR='sed -ie \"\$ a colocation colocation_$fs_rsc inf: $fs_rsc ms_$resource:Master\"' crm configure edit || exit 1
+EDITOR='sed -ie \"\$ a order order_$fs_rsc Mandatory: ms_$resource:promote $fs_rsc:start\"' crm configure edit || exit 1
+";
             }
             else {
-                assert_script_run "EDITOR=\"sed -ie '\$ a colocation colocation_$fs_rsc inf: $fs_rsc vg_$resource'\" crm configure edit", $default_timeout;
-                assert_script_run "EDITOR=\"sed -ie '\$ a order order_$fs_rsc Mandatory: vg_$resource $fs_rsc'\" crm configure edit", $default_timeout;
+                $edit_crm_config_script .= "
+EDITOR='sed -ie \"\$ a colocation colocation_$fs_rsc inf: $fs_rsc vg_$resource\"' crm configure edit || exit 1
+EDITOR='sed -ie \"\$ a order order_$fs_rsc Mandatory: vg_$resource $fs_rsc\"\' crm configure edit || exit 1
+";
             }
-
-            # Sometimes we need to cleanup the resource
-            rsc_cleanup $fs_rsc;
+            $clean_flag = "cleanup";
         }
 
+        # run bash script to edit crm configure
+        write_sut_file '/root/crm_edit_config.sh', $edit_crm_config_script;
+        assert_script_run 'cat /root/crm_edit_config.sh', $default_timeout;
+        assert_script_run 'bash /root/crm_edit_config.sh', $default_timeout;
```

Comment:
You can do here `bash -ex` when calling the script.

`-e` will make the script abort on any failure, so I think this will allow you to drop all of the calls to ` || exit 1` above. For example:
```
# cat > a.sh
echo hello
ls -l dont_exist
echo world
^D
# bash -e a.sh ; echo retval-$?
hello
ls: cannot access 'dont_exist': No such file or directory
retval-2
#
```
And `-x` will print commands as they're executed. Could be helpful to determine where the script fails in the cases where it does.

Reply:
good idea, thank you

Reply:
code was updated, and VRs worked well.